runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7370,7 +7371,7 @@ TR::CompilationInfoPerThreadBase::postCompilationTasks(J9VMThread * vmThread,
       {
       metaData = 0;
       }
-   else if (TR::CompilationInfo::shouldRetryCompilation(entry, _compiler))
+   else if (_compiler && TR::CompilationInfo::shouldRetryCompilation(entry, _compiler))
```

Comment:
This is a very rare case I ran into where `postCompilationTasks()` could be triggered twice. A steam failure exception might be thrown inside `postCompilationTasks()`, although the compilation is already successful at the server . This exception is caught in `CompilationInfoPerThreadBase::compile()` and `setCompilation(NULL)`  is called before `postCompilationTasks()` is called. 

However it only happens when sending free KOT messages. Since I'm going to remove this message, I plan to revert this change because we don't have other known cases to hit this path. 

```c++
CompilationInfoPerThreadBase::compile()
try 
{ 
	…
	postCompilationTasks();
}
catch 
{
	…
	postCompilationTasks();
}
```

https://github.com/eclipse/openj9/blob/ed3d30eedefeef319a1908a98556262ec8fa7c1e/runtime/compiler/control/CompilationThread.cpp#L7752-L7762.